server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader12to13.java

```diff
@@ -363,8 +364,18 @@ private void validateEmptyZKWorkerServerPaths(ServerContext context) {
         List<String> children = zr.getChildren(zkRoot + serverPath);
         for (String child : children) {
           if (child.contains(":")) {
-            throw new IllegalStateException("Found server address at " + serverPath + "/" + child
-                + ". Was expecting either a resource group name or nothing. Stop any referenced servers.");
+            String childPath = zkRoot + serverPath + "/" + child;
+            if (zr.getChildren(childPath).isEmpty()) {
+              // child is likely host:port and is an empty directory. Since there
+              // is no lock here, then the server is likely down (or should be).
+              // Remove the entry and move on.
+              context.getZooSession().asReaderWriter().recursiveDelete(childPath,
```

Comment:
I wonder if it would be safer if this upgrade code reserved its own lock on these empty nodes, to prevent other services from getting the lock while in this upgrade code.

Reply:
I guess deleting the lock is probably sufficient... no need to reserve it. I was just thinking of the case where a new one tried to come online.

Reply:
I don't know that a lock would help here. We know that it's an old path in ZooKeeper and that it's empty, so we are going to delete it. If another process for some reason is going to try and use it, we don't want that process up anyway, so deleting the server directory would delete any lock that is acquired after we determined the directory was empty, and deleting the lock would terminate the process.